pkg/detector/ospkg/ubuntu/testdata/fixtures/data-source.yaml

```diff
@@ -1,11 +1,11 @@
 - bucket: data-source
   pairs:
-    - key: ubuntu 20.04
+    - key: ubuntu 21.04
```

Comment:
The time is [fixed](https://github.com/aquasecurity/trivy/blob/633427de3a0d56336426292e12274486c0ed3250/pkg/detector/ospkg/ubuntu/ubuntu_test.go#L178) in the tests. Do we need to change the OS version?

Reply:
`time` doesnэt help for `ubuntu 20.04-ESM. 20.04 is not outdated` test.
We should take vulnerabilities from `20.04` for `20.04-ESM`, and we did it before `20.04-ESM` was added.
But now we take vulnerabilities from `20.04-ESM` directly (because map contains `20.04-ESM`)

https://github.com/aquasecurity/trivy/blob/312649c988eef4b9cc82d55a0447189497438b74/pkg/detector/ospkg/ubuntu/ubuntu.go#L140-L155

Reply:
Thanks for your explanation. I missed that there is an early return if the version is in the EOL list.

Since this only happens once every two years, it’s not a major burden. However, it might be a bit confusing that updating the EOL list causes tests to fail, especially since we (or only I) tend to forget the details over two years.
Is there a way to add new entries without modifying the existing tests? Maybe passing a custom eolDates?

Reply:
hm.. It looks asgood idea.
Added in https://github.com/aquasecurity/trivy/pull/8981/commits/07f066f1b4f668eefdf786cef7b5fd67a93a49f0